packages/wallet/ui/src/util/WalletBackendAdapter.js

```diff
@@ -130,15 +130,82 @@ export const makeWalletBridgeFromFollower = (
     ]),
   );
 
+  // We assume just one cosmos purse per brand.
+  const offers = {};
+  const brandToPurse = new Map();
+  const pursePetnameToBrand = new Map();
+
+  const updatePurses = () => {
+    const purses = [];
+    for (const [brand, purse] of brandToPurse.entries()) {
+      if (purse.currentAmount && purse.brandPetname) {
+        console.log(purse.currentAmount);
+        pursePetnameToBrand.set(purse.pursePetname, brand);
+        purses.push(purse);
+      }
+    }
+    // console.log(purses);
+    notifierKits.purses.updater.updateState(harden(purses));
+  };
+
   const followLatest = async () => {
-    for await (const { value: state } of iterateLatest(follower)) {
+    /** @type {number} */
+    let firstHeight;
+    for await (const { blockHeight } of iterateReverse(follower)) {
+      // TODO: Only set firstHeight and break if the value contains all our state.
```

Comment:
How do we tell when we get to a snapshot block?

Reply:
They don't exist yet, which is why we only have this comment to describe what the code should eventually do.